tmt/tests/booted/test-image-upgrade-reboot.nu

```diff
@@ -0,0 +1,70 @@
+# This test does:
+# bootc image copy-to-storage
+# podman build <from that image>
+# bootc switch <into that image> --apply
+# Verify we boot into the new image
+#
+use std assert
+use tap.nu
+
+const kargsv0 = ["testarg=foo", "othertestkarg", "thirdkarg=bar"]
+
+# This code runs on *each* boot.
+# Here we just capture information.
+bootc status
+let st = bootc status --json | from json
+let booted = $st.status.booted.image
+
+# Parse the kernel commandline into a list.
+# This is not a proper parser, but good enough
+# for what we need here.
+def parse_cmdline []  {
+    open /proc/cmdline | str trim | split row " "
+}
+
+# Run on the first boot
+def initial_build [] {
+    tap begin "local image push + pull + upgrade"
+
+    bootc image copy-to-storage
+
+    mkdir usr/lib/bootc/kargs.d
+    { kargs: $kargsv0 } | to toml | save usr/lib/bootc/kargs.d/05-testkargs.toml
+    # A simple derived container that adds a file, but also injects some kargs
+    "FROM localhost/bootc
+COPY usr/ /usr/
+" | save Dockerfile
+    # Build it
+    podman build -t localhost/bootc-derived .
+
+    # Now, switch into the new image
+    tmt-reboot -c "bootc switch --apply --transport containers-storage localhost/bootc-derived"
```

Comment:
This doesn't seem to work - perhaps TMT reboots at the end of the command if the command itself doesn't?

Maybe we do something like:

```
tmt-reboot -c "bootc switch --apply --transport containers-storage localhost/bootc-derived && exit 1"
```

In theory, if the `--apply` flag works, the `exit 1` should never be reached and the tests continue as expected.  
If the `exit 1` is hit, it should mean `--apply` didn't initiate a reboot, and the test will rightfully fail.

Thoughts?